runtime/lua/vim/util/buffer_manager.lua

```diff
@@ -0,0 +1,73 @@
+local vim = vim
+local api = vim.api
+
+-- config = {
+--   buf_init(bufnr, ...) -> lookup_value: any
+--   on_detach(bufnr, lookup_value)
+--   on_lines
+--   on_changedtick
+--   utf_sizes
+-- }
+-- Returns: { attach(bufnr, ...), iter(), get(bufnr) }
+--
+-- Any arguments passed after bufnr to attach() are forwarded to buf_init()
+-- on_lines, on_changedtick, utf_sizes are forwarded to nvim_buf_attach
+local function buffer_manager(config)
```

Comment:
First line of the docstring should say qualitatively what it actually does / what is the purpose.